packages/web/src/pages/settings-page/components/desktop/DeveloperApps/EditAppPage.tsx

```diff
@@ -0,0 +1,211 @@
+          name: values.name,
+          description: values.description
+        })
+      )
+      editDeveloperApp(values)
+    },
+    [editDeveloperApp, record]
+  )
+
+  const initialValues: DeveloperAppValues = {
+    userId,
+    apiKey: apiKey || '',
+    name: name || '',
+    description,
+    imageUrl
+  }
+
+  const isSubmitting = status !== Status.IDLE && status !== Status.ERROR
+
+  const copyApiKey = useCallback(() => {
+    if (!apiKey) return
+    copyToClipboard(apiKey)
+  }, [apiKey])
+
+  if (!params) return null
+
+  return (
+    <Formik
+      initialValues={initialValues}
+      onSubmit={handleSubmit}
+      validationSchema={toFormikValidationSchema(developerAppEditSchema)}
+    >
+      {() => (
+        <Form>
+          <Flex gap='m' direction='column'>
+            <Flex gap='m' alignItems='center'>
+              <Flex
+                borderRadius='xs'
+                css={{ overflow: 'hidden', alignSelf: 'stretch' }}
+                flex={'1 1 auto'}
+              >
+                <ImageField name='imageUrl' />
+              </Flex>
+              <Flex
+                flex={'1 1 auto'}
+                direction='column'
+                gap='m'
+              >
+                <TextField
+                  name='name'
+                  label={messages.appNameLabel}
+                  disabled={isSubmitting}
+                  maxLength={DEVELOPER_APP_NAME_MAX_LENGTH}
+                />
+                <TextField
+                  name='imageUrl'
+                  label={messages.imageUrlLabel}
+                  disabled={isSubmitting}
+                  maxLength={DEVELOPER_APP_IMAGE_URL_MAX_LENGTH}
+                />
+              </Flex>
+            </Flex>
+            <TextAreaField
+              name='description'
+              showMaxLength
+              maxLength={DEVELOPER_APP_DESCRIPTION_MAX_LENGTH}
+              disabled={isSubmitting}
+            />
+            <div className={styles.keyRoot}>
+              <span className={styles.keyLabel}>{messages.apiKey}</span>
+              <Divider type='vertical' className={styles.keyDivider} />
+              <span className={styles.keyText}>{apiKey}</span>
+              <Divider type='vertical' className={styles.keyDivider} />
+              <span>
+                <Toast text={messages.copied} mount={MountPlacement.PARENT}>
+                  <IconButton
+                    onClick={copyApiKey}
+                    aria-label={messages.copyApiKeyLabel}
+                    color='subdued'
+                    icon={IconCopy}
+                  />
+                </Toast>
+              </span>
+            </div>
+            <div className={styles.actionsContainer}>
+              <Button
+                variant='secondary'
+                type='button'
+                fullWidth
+                disabled={isSubmitting}
+                onClick={() => setPage(CreateAppsPages.YOUR_APPS)}
```

Comment:
I'm still on the fence about whether usage like this is better as a `useCallback`. Harmony button has a lot of internal logic already. So I doubt swapping the callback function on every render helps much. So I think I mostly want to convey a reminder to consider if the component might be memoized or might do expensive things on change of the onClick prop.
Probably fine here though 👍